src/anyio/_core/_fileio.py

```diff
@@ -114,3 +124,208 @@ async def open_file(file: Union[str, PathLike, int], mode: str = 'r', buffering:
+        return await to_thread.run_sync(self._path.is_char_device, cancellable=True)
+
+    async def is_dir(self) -> bool:
+        return await to_thread.run_sync(self._path.is_dir, cancellable=True)
+
+    async def is_fifo(self) -> bool:
+        return await to_thread.run_sync(self._path.is_fifo, cancellable=True)
+
+    async def is_file(self) -> bool:
+        return await to_thread.run_sync(self._path.is_file, cancellable=True)
+
+    async def is_mount(self) -> bool:
+        return await to_thread.run_sync(self._path.is_mount, cancellable=True)
+
+    def is_reserved(self) -> bool:
+        return self._path.is_reserved()
+
+    async def is_socket(self) -> bool:
+        return await to_thread.run_sync(self._path.is_socket, cancellable=True)
+
+    async def is_symlink(self) -> bool:
+        return await to_thread.run_sync(self._path.is_symlink, cancellable=True)
+
+    def iterdir(self) -> AsyncIterator['Path']:
+        gen = self._path.iterdir()
+        return _PathIterator(gen)
+
+    def joinpath(self, *args: Union[str, 'PathLike[str]']) -> 'Path':
+        return Path(self._path.joinpath(*args))
+
+    async def lchmod(self, mode: int) -> None:
+        await to_thread.run_sync(self.lchmod, mode)
+
+    async def lstat(self) -> os.stat_result:
+        return await to_thread.run_sync(self._path.lstat, cancellable=True)
+
+    async def mkdir(self, mode: int = 0o777, parents: bool = False,
+                    exist_ok: bool = False) -> None:
+        await to_thread.run_sync(self._path.mkdir, mode, parents, exist_ok)
+
+    async def open(self, mode: str = 'r', buffering: int = -1, encoding: Optional[str] = None,
+                   errors: Optional[str] = None, newline: Optional[str] = None) -> AsyncFile:
+        fp = await to_thread.run_sync(self._path.open, mode, buffering, encoding, errors, newline)
+        return AsyncFile(fp)
+
+    async def owner(self) -> str:
+        return await to_thread.run_sync(self._path.owner, cancellable=True)
+
+    async def read_bytes(self) -> bytes:
+        return await to_thread.run_sync(self._path.read_bytes)
+
+    async def read_text(self, encoding: Optional[str] = None, errors: Optional[str] = None) -> str:
+        return await to_thread.run_sync(self._path.read_text, encoding, errors)
+
+    def relative_to(self, *other: Union['Path', PathLike]) -> 'Path':
+        other = tuple((p._path if isinstance(p, Path) else p) for p in other)
+        return Path(self._path.relative_to(*other))
+
+    async def rename(self, target: Union[str, pathlib.PurePath, 'Path']) -> None:
+        if isinstance(target, Path):
+            target = target._path
+
+        await to_thread.run_sync(self._path.rename, target)
+
+    async def replace(self, target: Union[str, pathlib.PurePath, 'Path']) -> None:
+        if isinstance(target, Path):
+            target = target._path
+
+        await to_thread.run_sync(self._path.replace, target)
+
+    async def resolve(self) -> 'Path':
```

Comment:
resolve has a strict kwarg

Reply:
Yeah, I was gonna add that, just slipped my mind. But thanks for the reminder.